transport.go

```diff
@@ -134,3 +135,22 @@ func (t *Transport) refreshToken() error {
 
 	return nil
 }
+
+func addAcceptHeader(req *http.Request) {
```

Comment:
Could this be slightly rewritten to return early when possible, what's your thoughts on this? I haven't tested it at all.

```go
// addAcceptIfJSON adds the preview accept headers if the request does not already specify an
// accept header, or if it's already specifying acceptance of json type as defined by
// https://developer.github.com/v3/media/.
func addAcceptIfJSON(headers http.Header) {
	if headers.Get("Accept") == "" {
		headers.Set("Accept", acceptHeader)
                return
	}

	for _, header := range headers["Accept"] {
		if strings.HasSuffix(header, "json") {
			headers.Add("Accept", acceptHeader)
			return
		}
	}
}
```

Just accepting a `http.Header` also simplifies some of the tests, as the function just requires what it checks and modifies, nothing more. But again, I haven't tested this at all.

Reply:
@bradleyfalzon Made the changes